internal/commands/untrust_builder_test.go

```diff
@@ -0,0 +1,152 @@
+		logger        logging.Logger
+		outBuf        bytes.Buffer
+		tempPackHome  string
+		configPath    string
+		configManager configManager
+	)
+
+	it.Before(func() {
+		var err error
+
+		logger = ilogging.NewLogWithWriters(&outBuf, &outBuf)
+
+		tempPackHome, err = ioutil.TempDir("", "pack-home")
+		h.AssertNil(t, err)
+		h.AssertNil(t, os.Setenv("PACK_HOME", tempPackHome))
+
+		configPath = filepath.Join(tempPackHome, "config.toml")
+		configManager = newConfigManager(t, configPath)
+	})
+
+	it.After(func() {
+		h.AssertNil(t, os.Unsetenv("PACK_HOME"))
+		h.AssertNil(t, os.RemoveAll(tempPackHome))
+	})
+
+	when("#UntrustBuilder", func() {
+		when("no builder is provided", func() {
+			it("prints usage", func() {
+				cfg := configManager.configWithTrustedBuilders()
+				command := commands.UntrustBuilder(logger, cfg)
+				command.SetArgs([]string{})
+				h.AssertNil(t, command.Execute())
+				h.AssertContains(t, outBuf.String(), "Usage:")
+			})
+		})
+
+		when("builder is already trusted", func() {
+			it("removes builder from the config", func() {
+				builderName := "some-builder"
+
+				cfg := configManager.configWithTrustedBuilders(builderName)
+				command := commands.UntrustBuilder(logger, cfg)
+				command.SetArgs([]string{builderName})
+
+				h.AssertNil(t, command.Execute())
+
+				b, err := ioutil.ReadFile(configPath)
+				h.AssertNil(t, err)
+				h.AssertNotContains(t, string(b), builderName)
+
+				h.AssertContains(t,
+					outBuf.String(),
+					fmt.Sprintf("Builder %s is no longer trusted", style.Symbol(builderName)),
+				)
+			})
+
+			it("removes only the named builder when multiple builders are trusted", func() {
+				untrustBuilder := "stop/trusting:me"
+				stillTrustedBuilder := "very/safe/builder"
+
+				cfg := configManager.configWithTrustedBuilders(untrustBuilder, stillTrustedBuilder)
+				command := commands.UntrustBuilder(logger, cfg)
+				command.SetArgs([]string{untrustBuilder})
+
+				h.AssertNil(t, command.Execute())
+
+				b, err := ioutil.ReadFile(configPath)
+				h.AssertNil(t, err)
+				h.AssertContains(t, string(b), stillTrustedBuilder)
+				h.AssertNotContains(t, string(b), untrustBuilder)
+			})
+		})
+
+		when("builder wasn't already trusted", func() {
+			it("does nothing and reports builder wasn't trusted", func() {
+				neverTrustedBuilder := "never/trusted-builder"
+				stillTrustedBuilder := "very/safe/builder"
+
+				cfg := configManager.configWithTrustedBuilders(stillTrustedBuilder)
+				command := commands.UntrustBuilder(logger, cfg)
+				command.SetArgs([]string{neverTrustedBuilder})
+
+				h.AssertNil(t, command.Execute())
+
+				b, err := ioutil.ReadFile(configPath)
+				h.AssertNil(t, err)
+				h.AssertContains(t, string(b), stillTrustedBuilder)
+				h.AssertNotContains(t, string(b), neverTrustedBuilder)
+
+				h.AssertContains(t,
+					outBuf.String(),
+					fmt.Sprintf("Builder %s wasn't trusted", style.Symbol(neverTrustedBuilder)),
+				)
+			})
+		})
+	})
+}
+
+type configManager struct {
```

Comment:
Not particularly fond of a object to simply persist only two params. Seems like overkill for this particular case where a single function taking the arguments would be easier to understand and within context of it's usage.

Reply:
> Not particularly fond of a object to simply persist only two params

That's unfortunate. I'm super fond of anything that means I don't have to keep passing `t` into everything...

Reply:
I'm not trying to be particularly picky here as I can see this going in as-is. (The main blocker atm is [this](https://github.com/buildpacks/pack/pull/699/files#r442964759)). The consequences with this helper class is its added indirection albeit minor in this context. I'm purely making a statement that it seems like it's buying very little while adding some overhead in code and indirection.